ElectricCar.hpp

```diff
@@ -1,16 +1,18 @@
 #include "ElectricEngine.hpp"
+#include "Car.hpp"
+#include"Exceptions.hpp"
+#pragma once
 
-class ElectricCar
+class ElectricCar : virtual public Car
 {
 public:
     ElectricCar(ElectricEngine* engine);
-    ~ElectricCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
-    void charge();
+    virtual ~ElectricCar();
 
+    void charge();
+    ElectricEngine* changeEngine (int, int);
```

Comment:
to wam nie nadpisuje funkcji z klasy bazowej, tyko tworzy nową funkcję, ponieważ ma inny typ zwracany. Funkcja 
`void changeEngine (int, int);` z Car nie jest w ogóle używana. Można ją usunąć.